### ISSUE TITLE
Isolation of Execution Contexts

### ISSUE THREAD

**Issue by fantaclaus** [Microsoft/ChakraCore#2092]
I've executed the following code both with JSRT Legacy (jscript9.dll) and with the release version 1.3.0 of ChakraCore and have received very different results.

In case of version 1.3.0 the code has successfully completed.

In case of JSRT Legacy the code has thrown an exception of type `JavaScriptUsageException` with a message "Invalid argument." on the invocation of `obj.GetProperty`.

```c#
using (var runtime = JavaScriptRuntime.Create())
{
	JavaScriptValue obj;

	var context1 = runtime.CreateContext();
	var context2 = runtime.CreateContext();

	using (new JavaScriptContext.Scope(context1))
	{
		obj = JavaScriptValue.CreateObject();

		var propId = JavaScriptPropertyId.FromString("prop");
		obj.SetProperty(propId, JavaScriptValue.True, true);
	}

	using (new JavaScriptContext.Scope(context2))
	{
		var propId = JavaScriptPropertyId.FromString("prop");
		var v = obj.GetProperty(propId); // <- this fails in JSRT Legacy
	}
}
```

So I have a question: What is the intended behavior and which version of JSRT has a bug?

In other words: must the execution contexts be isolated or not?

**Comment by darfink** [comment]
I also stumbled upon this behaviour just now, and I am eager to know what is exactly isolated between different contexts.

**Comment by MikeHolman** [comment]
I guess we have made marshaling fields between contexts more permissive, but not really sure. I think this requires a bit of arcane knowledge.

@CurtisMan may be able to help clarify what's going on.

**Comment by curtisman** [comment]
@fantaclaus Yes.  In ChakraCore all versions, and Chakra shipped in Win10 November Update (2015) or after, we added support to allow object to be used across different `JsContext`  For all versions, JsContext from the same JsRuntime reside on the same GC heap anyways. For complete isolation, you can create separate `JsRuntime`

@darfink You are correct. `JsInstanceOf` is equivalent of the `instanceof` operator in script, and it is based on prototype chain equality, and thus has the same limitation.

**Comment by dilijev** [comment]
Looks like there is no compatibility concern because the question references legacy jscript9 behavior. Also looks like this question is answered. Closing.
